lib/npm/spawn-npm.js

```diff
@@ -4,10 +4,13 @@ const chalk = require('chalk');
 const execa = require('execa');
 const ora = require('ora');
 
-function install(packages, currentState) {
+function install(packages, currentState, installer) {
     if (!packages.length) {
         return Promise.resolve(currentState);
     }
+    if (typeof installer !== 'string') {
```

Comment:
this won't be needed if you use `state.js`'s `currentState` because all values have a default.